src/test/golang/test_netlink.go

```diff
@@ -0,0 +1,31 @@
+package main
+
+import (
+	"fmt"
+	"net"
+	"runtime"
+)
+
+func println(args ...interface{}) {
+	// Runtime error on go 1.17
+	// https://github.com/golang/go/issues/47873
+	if (runtime.Version() != "go1.17") {
+		fmt.Println(args...);
+	}
+}
```

Comment:
The go test is great, but it would also be nice to have a rust test that tests the syscall specifically with the shorter buffer. I think you could re-use your `flood` test to construct a `Nlmsghdr`, shrink the `nl_len` field, and call `sendto` with the smaller buffer.